netty/src/main/java/io/grpc/netty/NettyClientStream.java

```diff
@@ -54,7 +54,9 @@
  */
 class NettyClientStream extends AbstractClientStream {
   private static final InternalMethodDescriptor methodDescriptorAccessor =
-      new InternalMethodDescriptor(InternalKnownTransport.NETTY);
+      new InternalMethodDescriptor(
+          NettyClientTransport.class.getName().contains("grpc.netty.shaded")
+              ? InternalKnownTransport.NETTY_SHADED : InternalKnownTransport.NETTY);
```

Comment:
I'm not sure if it's possible that the compiler can optimize this `contains()?:` expression and replace it with `InternalKnownTransport.NETTY` directly before shadow plugin does its job. My local interop test did not show it happened.

Reply:
No, it won't be able to rework this string. You could also just do `"shaded"` as another option. But all those variations should be fine.

Reply:
Just "shaded" will produce false positive if grpc as a whole is shaded by the application.